File: assets/code-variants/ch12/22.py
def test_should_allow_multiple_users():
    user1 = new_user(state=State.NORMAL)
    user2 = new_user(state=State.NORMAL)

    forum = Forum()
    forum.register(user1)
    forum.register(user2)

    assert forum.has_registered_user(user1)
    assert forum.has_registered_user(user2)

def test_should_not_register_banned_users():
    user = new_user(state=State.BANNED)

    forum = Forum()
    try:
        forum.register(user)
    except BannedUserError:
        pass

    assert not forum.has_registered_user(user)

File: assets/code-variants/ch12/22.java
@Test
public void shouldAllowMultipleUsers() {
    User user1 = newUser().setState(State.NORMAL).build();
    User user2 = newUser().setState(State.NORMAL).build();

    Forum forum = new Forum();
    forum.register(user1);
    forum.register(user2);

    assertThat(forum.hasRegisteredUser(user1)).isTrue();
    assertThat(forum.hasRegisteredUser(user2)).isTrue();
}

@Test
public void shouldNotRegisterBannedUsers() {
    User user = newUser().setState(State.BANNED).build();

    Forum forum = new Forum();
    try {
        forum.register(user);
    } catch(BannedUserException ignored) {}

    assertThat(forum.hasRegisteredUser(user)).isFalse();
}
